### PR TITLE
[Ember-Migration] update:ember changes for main-section done

### LINKED ISSUE #715: [Ember-Migration] Main Section
### Requirements
As we are migrating towards Emberjs we need the given section to be sync with our production, to look it exactly like production version. In this ticket we have to make the mentioned section UI as well as functionalities to match same as in production version.

#### In vanilla site (currently running www site in production, created with plain HTML, CSS, JS)
image

#### In beta site(created with EmberJS)
image

### Important links

- https://beta.realdevsquad.com/ (This one is made with Ember js)
- https://www.realdevsquad.com/ (This one is the production site)

### Additional Note

- Design should be same as in prod.
- Take reference from the prod site wherever needed.
- It should be responsive.
- Code quality should be maintained.

### Questions you can have in mind

1. Is this should be under feature flag?
Yes, it should be under feature dev=true

2. What is feature flag?
It's a simple variable which we can use to turn off/on the feature. Currently we use dev=true in URL query.

3. Do you have to use the same code as in production site?
No, the solution is upto you how you want to approach the problem. It should be same UI wise and functionality wise, rest other implementation is on you.

4. In which branch ember code is in?
It's in `develop-ember` branch, make sure to checkout to that branch.

### PR DESCRIPTION
**Pull request by devananth**
### What is the change?

- Synced beta site with main-section with vanilla site main-section
- Migration changes are under `?dev=true` feature flag

### Is Development Tested?

- [x] Yes
- [ ] No

image

### Before / After Change Screenshots

Before :
image

After :
image

### Video of Changes
https://www.loom.com/share/ab9ecdc31e2b409c8a3dad61bf819fe6?sid=328272c7-51c6-46d8-b84e-ab278e0cc383

**Note :**
- The changes are under ?dev=true feature flag

**Follow Up:**
- The feature flag should be removed from these changes after migration changes are confirmed.

### CODE REVIEW

File: app/styles/header.module.css [hunk 1]
```diff
@@ -164,4 +164,77 @@
...
+}
+
+.social__icon:hover{
+  transform: scale(1.25);
+}
+
+.social__seperators-vertical--dev {
+  margin: 0px 10px;
+  color : var(--color-black);
+
```
[the remark this answers is not in the document]
Comment (devananth) [reply app/styles/header.module.css:227]: removed the space

File: app/components/header.hbs
[the remark this answers is not in the document]
Comment (devananth) [reply app/components/header.hbs:1]: changed the classnames according to BEM convention.

File: app/components/header.hbs [hunk 2]
```diff
@@ -1,3 +1,40 @@
+{{#if this.isDevMode}}
+<section aria-label='header under dev feature flag' class='header--dev'>
```
[the remark this answers is not in the document]
Comment (devananth) [reply app/components/header.hbs:2]: changed `header` to `main`

File: public/assets/images/hero-community.png
[the remark this answers is not in the document]
Comment (devananth) [reply public/assets/images/hero-community.png:1]: Yes, it is available in `.png` format took this `img` from production site.

File: tests/acceptance/migration-changes-under-feature-flag-test.js
[the remark this answers is not in the document]
Comment (devananth) [reply tests/acceptance/migration-changes-under-feature-flag-test.js:1]: Added additional tests for all the elements under the `feature flag`

File: app/styles/header.module.css [hunk 3]
```diff
@@ -164,4 +164,77 @@
...
+}
+
+@media (min-width: 420px){
```
[the remark this answers is not in the document]
Comment (devananth) [reply app/styles/header.module.css]: Changed to breakpoints already used in the app

File: yarn.lock
[the remark this answers is not in the document]
Comment (devananth) [reply yarn.lock:1]: Not sure about the changes. But deleted the file and downloaded from repo and pushed changes.

File: app/components/header.hbs [hunk 4]
```diff
@@ -1,3 +1,40 @@
...
+<section aria-label='main-section under dev feature flag' class='main--dev'>
+    <h1 data-test-main-welcome-title class="main__welcome--title">Welcome to Real Dev Squad</h1>
+    <img
+      data-test-main-hero-img
+      class='main__hero--img'
+      src='assets\images\hero-community.png'
+      alt='Real Dev Squad'
+    />
+    <div data-test-main-container class='main__container'>
+        <div class='social__links'>
```
Comment (satyam73) [inline app/components/header.hbs:11]: there's no block named social, please follow BEM convention

File: app/styles/header.module.css [hunk 5]
```diff
@@ -164,4 +164,76 @@
...
+  width: 400px;
+  height: 400px;
+  display: block;
+  margin: 0 auto;
+}
+
+.social__links{
+  text-align: center;
+  font-weight: 400;
+  font-size: 1.125em;
```
Comment (satyam73) [inline app/styles/header.module.css:197]: Use rem for fonts

File: app/styles/header.module.css [hunk 6]
```diff
@@ -164,4 +164,76 @@
...
+  transform: scale(1.25);
+}
+
+.social__seperators-vertical--dev {
+  margin: 0px 10px;
+  color : var(--color-black);
+}
+
+@media (max-width: 425px){
+
```
Comment (satyam73) [inline app/styles/header.module.css:230]: NIT: extra spacin here

File: app/components/header.hbs (hunk 4 above)
Comment (satyam73) [inline app/components/header.hbs:11]: there's no block named social, please follow BEM convention

File: app/styles/header.module.css (hunk 5 above)
Comment (satyam73) [inline app/styles/header.module.css:197]: use rem for fonts

File: app/styles/header.module.css
Comment (satyam73) [inline app/styles/header.module.css:1]: use rem units please so that it is consistent throughout the code

**Review by satyam73 — CHANGES_REQUESTED** [review]

File: app/styles/header.module.css (hunk 6 above)
Comment (devananth) [reply app/styles/header.module.css]: removed the space.

File: app/components/header.hbs (hunk 4 above)
Comment (devananth) [reply app/components/header.hbs]: changed, added a `block social`.

File: app/styles/header.module.css (hunk 5 above)
Comment (devananth) [reply app/styles/header.module.css]: changed to rem.

File: app/styles/header.module.css
Comment (devananth) [reply app/styles/header.module.css:1]: Replaced atmost all `px` to `rem`

**Review by satyam73 — APPROVED** [review]

**Review by shubhamsigdar1 — APPROVED** [review]

**Pull request merged**